x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/Exporters.java

```diff
@@ -225,7 +233,7 @@ private void doExport(final ExportBulk bulk, final Collection<MonitoringDoc> doc
         } catch (ExportException e) {
             exceptionRef.set(e);
         } finally {
-            bulk.close(lifecycleState() == Lifecycle.State.STARTED, ActionListener.wrap(r -> {
```

Comment:
there are alot of states here, so i am kinda getting lost... but i _think_ this different then before since this can now flush if the Lifecycle.State.STARTED == false. Not sure if this an issue

Reply:
You're correct. This is a subtle change in behavior, but it was always an unnecessary one. The clients used by the exporter implementations will be closed when the process shuts down even if we're in the middle of a flush / close, and the `exporter` don't exist until the lifecycle is started, which means we couldn't have even gotten here anyway.